lib/puppet.rb

```diff
@@ -36,6 +50,29 @@ module Puppet
   require 'puppet/environments'
 
   class << self
+    if GETTEXT_AVAILABLE
+      # TODO: Handle the rubygems case?
+      local_locale_path = File.absolute_path('../locales', File.dirname(__FILE__))
+      posix_system_locale_path = File.absolute_path('../../../share/puppet/locale', File.dirname(__FILE__))
+      win32_system_locale_path = File.absolute_path('../../../../../puppet/share/puppet/locale', File.dirname(__FILE__))
+
+      if File.exist?(local_locale_path)
+        locale_path = local_locale_path
+      elsif File.exist?(win32_system_locale_path)
+        locale_path = win32_system_locale_path
+      elsif File.exist?(posix_system_locale_path)
+        locale_path = posix_system_locale_path
+      else
+        # We couldn't load our locale data. Possibly we're loaded as a rubygem or something?
+        locale_path = nil
+      end
+
+      if locale_path
+        GettextSetup.initialize(locale_path)
+        FastGettext.locale = GettextSetup.negotiate_locale(ENV["LANG"])
```

Comment:
@glennsarti we've already pulled in (and shipped in 1.8.0) the `locale` gem (mentioned above - https://github.com/puppetlabs/puppet-agent/commit/29e90b9375c23ee4bd854ad720188bea13f65754). Are you suggesting we remove the gem?

Maybe we could still use it and contribute improvements (ie using the newer API) upstream rather than maintaining our own logic? 

Then we also don't necessarily need to expose another API for locale within puppet and can avoid augmenting `Puppet::Util`..